### PR TITLE
Update for qb-target and polyzone

### PR DESCRIPTION
**Pull request by deandum**
To enable `qb-target` for apartments, change `Apartments.UseTarget` to `true` in `config.lua`.

### CODE REVIEW

**Review by iktray — APPROVED** [review]

**Comment by BerkieBb** [comment]
It's not good to register target exports in a loop

**Comment by deandum** [comment]
> It's not good to register target exports in a loop

There is a `created` field on each zone created so that it is only initiated once. The loop is used to update the targets if the player moves to a new apartment, but the zones are created just once.

**Comment by BerkieBb** [comment]
I see

**Review by BerkieBb — APPROVED** [review]

**Comment by DannysRP** [comment]
Not sure if this is a issue with apartments or target but I've heard from others that the target option does not show up at the exit if you are still using the gabz default apartment.

**Comment by deandum** [comment]
> Not sure if this is a issue with apartments or target but I've heard from others that the target option does not show up at the exit if you are still using the gabz default apartment.

qb-target is not just a plug-and-play mechanic. The PZ has to be on top of an entity to work well, otherwise, there is nothing to target (that's how raycasting works in this game). With that being said, I don't know what `gabz default apartment` is, but you might just need to update the coords and the offset in the config file to match with your needs.

**Comment by DannysRP** [comment]
This was an issue other people have stated in support within the QBCore discord. Gabz pink cage apartment used the be the default apartment on all QBCore builds until it got updated a while back to the new starting apartment. Some people still use the old apartment and were having said issue I was speaking of

**Comment by DannysRP** [comment]
Small issue. When you leave your apartment after creating a new character the apartment says "Move Here" instead of "Enter Apartment". Only way to fix is either leave and join back or restart qb-apartments. Only noticed this with target config

**Comment by deandum** [comment]
> Small issue. When you leave your apartment after creating a new character the apartment says "Move Here" instead of "Enter Apartment". Only way to fix is either leave and join back or restart qb-apartments. Only noticed this with target config

Isn't it random which apartment is assigned to you? It is possible that this can happen, but I don't think it is necessarily an issue.

**Comment by DannysRP** [comment]
No you choose an apartment upon your time entering the server. Then when you leave and try to enter it says "Move Here" instead of "Enter Apartment"

**Comment by deandum** [comment]
> No you choose an apartment upon your time entering the server. Then when you leave and try to enter it says "Move Here" instead of "Enter Apartment"

Ok... try again

**Comment by DannysRP** [comment]
When I get a chance I'll check it out and let you know if I see any other issues

**Review by BerkieBb — APPROVED** [review]

**Review by ItsANoBrainer — CHANGES_REQUESTED** [review]
LGTM after these requested changes. Tested both target and menu of everything.

File: client/main.lua [hunk 1]
```diff
@@ -196,28 +457,118 @@ function MenuOwners()
...
+
+    headerMenu[#headerMenu+1] = {
+        header = Lang:t('text.close_menu'),
+        txt = "",
+        params = {
+            event = "qb-menu:client:closeMenu"
+        }
+    }
+
+    exports['qb-menu']:showHeader(headerMenu)
```
[the remark this answers is not in the document]
Comment (deandum) [reply client/main.lua:527]: Thanks! Copy paste 🤦‍♂️

File: client/main.lua [hunk 2]
```diff
@@ -196,28 +457,118 @@ function MenuOwners()
...
```
[the remark this answers is not in the document]
Comment (deandum) [reply client/main.lua]: Copy paste 🤦‍♂️

**Review by Muri22 — APPROVED** [review]

File: client/main.lua [hunk 3]
```diff
@@ -160,7 +421,7 @@ function MenuOwners()
     QBCore.Functions.TriggerCallback('apartments:GetAvailableApartments', function(apartments)
         if next(apartments) == nil then
             QBCore.Functions.Notify(Lang:t('error.nobody_home'), "error", 3500)
-            closeMenuFull()
+            CloseMenuFull()
         else
             local vehicleMenu = {
```
[the remark this answers is not in the document]
Comment (ItsANoBrainer) [reply client/main.lua]: Theres more to change than just this line, the rest of the references below also use vehicleMenu and need to be changed

[the remark this answers is not in the document]
Comment (deandum) [reply client/main.lua]: Yeah, I actually updated those, but never pushed the changes until now. It should be good

**Review by ItsANoBrainer — APPROVED** [review]

**Pull request merged**